diff --git a/src/gmpy2.c b/src/gmpy2.c
index b2f0eac..f10247b 100644
--- a/src/gmpy2.c
+++ b/src/gmpy2.c
@@ -909,7 +909,7 @@ PyMODINIT_FUNC initgmpy2(void)
 
     /* Validate the sizes of the various typedef'ed integer types. */
 
-#if defined _WIN64 && (MPIR || MSYS2)
+#if defined _WIN64 && (MPIR || MSYS2) && 0
     if (sizeof(mp_bitcnt_t) != sizeof(PY_LONG_LONG)) {
         /* LCOV_EXCL_START */
         SYSTEM_ERROR("Size of PY_LONG_LONG and mp_bitcnt_t not compatible (_WIN64 && MPIR)");
