diff --git a/setup.py b/setup.py
index f807404..96ec403 100644
--- a/setup.py
+++ b/setup.py
@@ -254,7 +254,7 @@ class gmpy_build_ext(build_ext):
         # Add the static linking options.
 
         # Add MSVC specific options.
-        if windows and not msys2:
+        if False:
             self.extensions[0].extra_link_args.append('/MANIFEST')
             self.extensions[0].define_macros.append(("MPIR", 1))
             if not static:
diff --git a/src/gmpy2_convert_gmp.c b/src/gmpy2_convert_gmp.c
index 6447397..186419d 100644
--- a/src/gmpy2_convert_gmp.c
+++ b/src/gmpy2_convert_gmp.c
@@ -194,7 +194,7 @@ GMPy_PyLong_From_MPZ(MPZ_Object *obj, CTXT_Object *context)
     assert(CHECK_MPZANY(obj));
 
     /* Assume gmp uses limbs as least as large as the builtin longs do */
-    assert(mp_bits_per_limb >= SHIFT);
+    assert(mp_bits_per_limb >= PyLong_SHIFT);
 
     if (mpz_sgn(obj->z) < 0) {
         negative = 1;
